airbyte-scheduler/src/main/java/io/airbyte/scheduler/client/DefaultSchedulerJobClient.java

```diff
@@ -73,26 +74,37 @@ public Job createGetSpecJob(String dockerImage) throws IOException {
   }
 
   @Override
-  public Job createSyncJob(
-                           SourceConnection source,
-                           DestinationConnection destination,
-                           StandardSync standardSync,
-                           String sourceDockerImage,
-                           String destinationDockerImage)
+  public Job createOrGetActiveSyncJob(SourceConnection source,
+                                      DestinationConnection destination,
+                                      StandardSync standardSync,
+                                      String sourceDockerImage,
+                                      String destinationDockerImage)
       throws IOException {
-    final long jobId = jobCreator.createSyncJob(
+    final Optional<Long> maybeJobId = jobCreator.createSyncJob(
         source,
         destination,
         standardSync,
         sourceDockerImage,
         destinationDockerImage);
+
+    long jobId = maybeJobId.isEmpty()
+        ? jobPersistence.getLastSyncScope(standardSync.getConnectionId()).orElseThrow(() -> new RuntimeException("No job available")).getId()
+        : maybeJobId.get();
+
     return waitUntilJobIsTerminalOrTimeout(jobId);
   }
 
   @Override
-  public Job createResetConnectionJob(DestinationConnection destination, StandardSync standardSync, String destinationDockerImage)
+  public Job createOrGetActiveResetConnectionJob(DestinationConnection destination,
+                                                 StandardSync standardSync,
+                                                 String destinationDockerImage)
       throws IOException {
-    final long jobId = jobCreator.createResetConnectionJob(destination, standardSync, destinationDockerImage);
+    final Optional<Long> maybeJobId = jobCreator.createResetConnectionJob(destination, standardSync, destinationDockerImage);
+
+    long jobId = maybeJobId.isEmpty()
+        ? jobPersistence.getLastSyncScope(standardSync.getConnectionId()).orElseThrow(() -> new RuntimeException("No job available")).getId()
+        : maybeJobId.get();
+
     return waitUntilJobIsTerminalOrTimeout(jobId);
```

Comment:
These don't seem right for reset & sync. do you confirm?

Reply:
It does seem off to wait until the whole sync job is complete. Would prefer returning a boolean to indicate if the job was queued. If returned false, the UI can also display a message "A job is already in progress".

Reply:
Separately we should allow canceling a job in case it's stuck. But this can be done in a following PR.

Reply:
> These don't seem right for reset & sync. do you confirm?

confirmed.